### PR TITLE
Upstream some LDC patches

### PR DESCRIPTION
**Pull request by kinke**
Pinging @klickverbot and @ibuclaw.

### CODE REVIEW

File: src/core/sys/posix/pwd.d [hunk 1]
```diff
@@ -196,10 +196,14 @@ else version( OpenBSD )
     int getpwnam_r(in char*, passwd*, char*, size_t, passwd**);
     int getpwuid_r(uid_t, passwd*, char*, size_t, passwd**);
 }
-else version (Solaris)
+else version( Solaris )
```
Comment (ibuclaw) [inline src/core/sys/posix/pwd.d]: Using this coding style is discouraged. I should probably get round to mechanically changing it to the right style.

**Comment by kinke** [comment]
Ping - should be a straightforward merge except for the AArch64 assembly, but that's all experimental anyway (and surely better than nothing, i.e., the status quo).

File: src/core/threadasm.S [hunk 2]
```diff
@@ -449,4 +449,59 @@ fiber_switchContext:
...
+ *    9: x29 (fp)  <-- oldp / *newp save stack top
+ *    8: x30 (lr)
+ *    7: d8
+ *   ...
+ *    0: d15       <-- sp
+ */
+        .text
+        .global CSYM(fiber_switchContext)
+        .p2align  2
+CSYM(fiber_switchContext):
```
Comment (ibuclaw) [inline src/core/threadasm.S]: @jpf91 please check this.

Comment (jpf91) [reply src/core/threadasm.S]: AFAICS this should be correct 👍

**Pull request merged**